lunatrace/bsl/hasura/migrations/lunatrace/1652400983913_add-pr-comment-disable/up.sql

```diff
@@ -1,60 +1,14 @@
-ALTER TABLE public.settings ADD COLUMN pr_feedback_enabled boolean DEFAULT true NOT NULL;
 ALTER TABLE public.settings DROP COLUMN is_org_settings;
-ALTER TABLE public.projects ADD CONSTRAINT project_settings_fkey FOREIGN KEY (settings_id) REFERENCES public.settings;
-ALTER TABLE public.organizations ADD CONSTRAINT organization_settings_fkey FOREIGN KEY (settings_id) REFERENCES public.settings;
+ALTER TABLE projects DROP COLUMN settings_id;
+ALTER TABLE organizations DROP COLUMN settings_id;
 
---  get caught up on the missing settings rows
-DO $$
-    declare
-        project record;
-    BEGIN
-FOR project IN SELECT * FROM public.projects WHERE settings_id IS NULL
-LOOP
-    WITH setting AS (INSERT INTO public.settings DEFAULT VALUES RETURNING id)
-    UPDATE public.projects
-    SET settings_id=setting.id
-    FROM setting
-    WHERE public.projects.id = project.id;
-END LOOP;
-END; $$;
+ALTER TABLE public.settings ADD COLUMN pr_feedback_disabled boolean DEFAULT false;
```

Comment:
I don't think this follows the "nullable" pattern that we talked about on slack.  The part of having it default to null that I like  is that we can detect when a user never set the column, that way we can change the default if necessary or override unset settings from a higher-order group such as at the organization level.  

Treating the entire settings row as nullable isn't the right abstraction to achieve that, since it will only tell us if ANY setting has been changed, not if a specific setting has been set. To achieve what we talked about, I would make the column `boolean DEFAULT NULL;` if you want to be explicit. 

Having the entire row be nullable is just an inconvenient thing we need to code around in our feature code.

Reply:
I don't think this follows the "nullable" pattern that we talked about on slack.  The part of having it default to null that I like  is that we can detect when a user never set the column, that way we can change the default if necessary or override unset settings from a higher-order group such as at the organization level.  

Treating the entire settings row as nullable isn't the right abstraction to achieve that, since it will only tell us if ANY setting has been changed, not if a specific setting has been set. To achieve what we talked about, I would make the column `boolean DEFAULT NULL;` if you want to be explicit. 

Having the entire row be nullable is just an inconvenient thing we need to code around in our feature code.